packages/nx/src/tasks-runner/life-cycles/task-history-life-cycle.ts

```diff
@@ -0,0 +1,56 @@
+import { Task } from '../../config/task-graph';
+import { output } from '../../utils/output';
+import {
+  getHistoryForHashes,
+  TaskRun,
+  writeTaskRunsToHistory as writeTaskRunsToHistory,
+} from '../../utils/task-history';
+import { LifeCycle, TaskResult } from '../life-cycle';
+
+export class TaskHistoryLifeCycle implements LifeCycle {
+  private startTimings: Record<string, number> = {};
+  private taskRuns: TaskRun[] = [];
+
+  startTasks(tasks: Task[]): void {
+    for (let task of tasks) {
+      this.startTimings[task.id] = new Date().getTime();
+    }
+  }
+
+  async endTasks(taskResults: TaskResult[]) {
+    const taskRuns: TaskRun[] = taskResults.map((taskResult) => ({
+      project: taskResult.task.target.project,
+      target: taskResult.task.target.target,
+      configuration: taskResult.task.target.configuration,
+      hash: taskResult.task.hash,
+      code: taskResult.code,
+      status: taskResult.status,
+      start: taskResult.task.startTime ?? this.startTimings[taskResult.task.id],
+      end: taskResult.task.endTime ?? new Date().getTime(),
+    }));
+    this.taskRuns.push(...taskRuns);
+  }
+
+  async endCommand() {
+    await writeTaskRunsToHistory(this.taskRuns);
+    const history = await getHistoryForHashes(this.taskRuns.map((t) => t.hash));
+    const flakyTasks: string[] = [];
+
+    // check if any hash has different exit codes => flaky
+    for (let hash in history) {
+      if (
+        history[hash].length > 1 &&
+        history[hash].some((run) => run.code !== history[hash][0].code)
+      ) {
+        flakyTasks.push(
+          `${history[hash][0].project}:${history[hash][0].target}`
```

Comment:
There should be some util to create this kind of task name. This is missing the confguration.